src/main/java/frc/robot/CommandFactory.java

```diff
@@ -55,4 +78,20 @@ public Command goToCoralStationAndIntake() {
                         Constants.ArmPositions.CORAL_STATION_WRIST_ANGLE)
                 .andThen(grabber.withVoltageUntilDetected(-1.0));
     }
+
+    public Command alignToReef(RobotMode robotMode){
+        return Commands.parallel(superStructure.followPositions(() -> robotMode),
+                Commands.repeatingSequence(
+                        Commands.defer(() -> drivetrain.pathToReefPoseCommand(getClosestBranch(drivetrain.getPose())), Set.of(drivetrain))));
+    }
+
+    public Command alignToCoralStation(RobotMode robotMode){
+         return Commands.parallel(this.goToCoralStationAndIntake(),
+                Commands.repeatingSequence(
+                        Commands.defer(() -> drivetrain.pathToCoralStationPoseCommand(getClosestCoralStationPosition(
+                                drivetrain.getPose()
+                        )), Set.of(drivetrain)))
+        );
+    }
```

Comment:
I think these methods should take in a `Supplier<RobotMode>`. Remember, these methods are only executed once one the command is bound to a button.